app/Database/CourseQueries.hs

```diff
@@ -144,8 +144,10 @@ buildMeetTimes meet = do
 
 -- | Looks up a graph using its title then gets the Shape, Text and Path elements
 -- for rendering graph (returned as JSON).
-getGraphJSON :: T.Text -> IO Response
-getGraphJSON graphName = getGraph graphName >>= withDefault
+getGraphJSON :: ServerPart Response
+getGraphJSON = do
+    graphName <- lookText' "graphName" 
+    liftIO $ getGraph graphName >>= withDefault
```

Comment:
Now that you've introduced using `do` notation, please change the `>>=` in here as well